pkg/pipeline/write/write_loki.go

```diff
@@ -231,7 +231,8 @@ func (l *Loki) processRecords() {
 			log.Debugf("exiting writeLoki because of signal")
 			return
 		case record := <-l.in:
-			err := l.ProcessRecord(record)
+			// copy record before process to avoid alteration on parallel stages
+			err := l.ProcessRecord(record.Copy())
```

Comment:
@jpinsonneau I suggest moving the copy process to be inside the `ProcessRecord` function --- the rule should be that we are not allowed to change the inputs --- and we need to maybe check also in other places that we are not doing that.

Reply:
@KalmanMeth ... so should we add copy also there ??? do you want to open a PR for that?